pkg/plugins/manager/pipeline/initialization/initialization.go

```diff
@@ -48,13 +48,30 @@ func (i *Initialize) Initialize(ctx context.Context, ps []*plugins.Plugin) ([]*p
 	var err error
 	initializedPlugins := make([]*plugins.Plugin, 0, len(ps))
 	for _, p := range ps {
+		initializeSteps := i.initializeSteps
+		if p.Status.Errored {
+			// Plugin already failed in a previous stage, we only want to register it.
+			initializeSteps = []InitializeFunc{}
+			for _, init := range i.initializeSteps {
+				if isRegistrationFunc(init) {
+					initializeSteps = []InitializeFunc{init}
+					break
+				}
+			}
+			if len(initializeSteps) == 0 {
+				return ps, nil
+			}
+		}
```

Comment:
This is the weird part, stages generally should skip its steps if the plugin failed before but here we want to still register it, which is an initialization step. Since there is no name associated to the step, my only option is to dinamically check the name of the function at runtime (see `isRegistrationFunc` definition below).

Another option would be to change the `initializeSteps` from functions to interfaces, so we could have two different functions (something like `Step()` and `Name()`) but that would require to change all the existing steps. WDYT?